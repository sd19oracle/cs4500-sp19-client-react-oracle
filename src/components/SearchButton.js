import React from 'react'

function SearchButton(props) {
    return (
        <button onClick={props.toggleSearch}>
            {props.searchButtonOn ? 'Search' : 'Clear Search'}
        </button>
    );
}

export default SearchButton