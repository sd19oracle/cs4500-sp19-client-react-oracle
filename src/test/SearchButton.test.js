import React from 'react'
import TestRenderer from 'react-test-renderer';
import SearchButton from '../components/SearchButton'

// snapshot testing for the search button when searchButtonOn
// is set to true
test('search button initial rendering', () => {
    const testRenderer = TestRenderer.create(
        <SearchButton searchButtonOn={true}/>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot()
});

// snapshot testing for the search button when searchButtonOn
// is set to false
test('search button clear search rendering', () => {
    const testRenderer = TestRenderer.create(
        <SearchButton searchButtonOn={false}/>
    );
    let tree = testRenderer.toJSON();
    expect(tree).toMatchSnapshot()
});

