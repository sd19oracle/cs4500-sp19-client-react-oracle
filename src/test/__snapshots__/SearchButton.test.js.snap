// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search button clear search rendering 1`] = `
<button>
  Clear Search
</button>
`;

exports[`search button initial rendering 1`] = `
<button>
  Search
</button>
`;
